src/uct/api/uct_def.h

```diff
@@ -210,6 +210,176 @@ typedef struct uct_cm_remote_data {
+ * @ref uct_cm_listener_conn_req_cb_handle are present, for backward compatibility
+ * support.
+ */
+enum uct_cm_listener_conn_req_cb_handle_field {
+    /** Enables @ref uct_cm_listener_conn_req_cb_handle::local_dev_name */
+    UCT_CM_LISTENER_CONN_REQ_CB_HANDLE_LOCAL_DEV_NAME = UCS_BIT(0),
+
+    /** Enables @ref uct_cm_listener_conn_req_cb_handle::conn_request */
+    UCT_CM_LISTENER_CONN_REQ_CB_HANDLE_CONN_REQUEST   = UCS_BIT(1),
+
+    /** Enables @ref uct_cm_listener_conn_req_cb_handle::remote_data */
+    UCT_CM_LISTENER_CONN_REQ_CB_HANDLE_REMOTE_DATA    = UCS_BIT(2)
+};
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Arguments to the listener's connection request callback.
+ *
+ * The local device name, connection request handle and the data the client sent.
+ * Used with the client-server API on a connection manager.
+ */
+typedef struct uct_cm_listener_conn_req_cb_handle {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_cm_listener_conn_req_cb_handle_field.
+     * Fields not specified by this mask will be ignored.
+     */
+    uint64_t                   field_mask;
+
+    /**
+     * Device name which handles the incoming connection request.
+     */
+    const char                 *local_dev_name;
+
+    /**
+     * Connection request handle. Can be passed to this callback from the
+     * transport and will be used by it to accept or reject the connection
+     * request from the client.
+     */
+    uct_conn_request_h         conn_request;
+
+    /**
+     * Remote data from the client.
+     */
+    const uct_cm_remote_data_t *remote_data;
+} uct_cm_listener_conn_req_cb_handle_t;
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Client-Server private data pack callback arguments field mask.
+ *
+ * The enumeration allows specifying which fields in
+ * @ref uct_sockaddr_priv_data_pack_cb_handle are present, for backward
+ * compatibility support.
+ */
+enum uct_sockaddr_priv_data_pack_cb_handle_field {
+    /** Enables @ref uct_sockaddr_priv_data_pack_cb_handle::dev_name */
+    UCT_SOCKADDR_PRIV_DATA_PACK_HANDLE_DEVICE_NAME = UCS_BIT(0)
+};
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Arguments to the client-server private data pack callback.
+ *
+ * Used with the client-server API on a connection manager.
+ */
+typedef struct uct_sockaddr_priv_data_pack_cb_handle {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_cm_ep_server_connect_cb_handle_field.
+     * Fields not specified by this mask will be ignored.
+     */
+    uint64_t                   field_mask;
+
+    /**
+     * Device name. This routine may fill the user's private data according to
+     * the given device name. The device name that is passed to this routine,
+     * corresponds to the dev_name field inside @ref uct_tl_resource_desc_t as
+     * returned from @ref uct_md_query_tl_resources.
+     */
+    const char                 *dev_name;
```

Comment:
let's use "dev_name[UCT_DEVICE_NAME_MAX]" to be consistent with uct_tl_device_resource_t